test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/FormatterTest.scala

```diff
@@ -321,11 +331,16 @@ class FormatterTest {
 
   @Test def `should_provide_o_conversion`(): Unit = {
     assertF("10", "%o", 8)
+    assertF("52", "%o", new BigInteger("42"))
 
     assertF("00020", "%05o", 16)
     assertF("37777777766", "%5o", -10)
     assertF("37777777766", "%05o", -10)
 
+    assertF("00020", "%05o", new BigInteger("16"))
+    assertF("  -12", "%5o", new BigInteger("-10"))
+    assertF("-0012", "%05o", new BigInteger("-10"))
```

Comment:
Hum, apparently this test does not pass: we get `00-12` instead of the correct `-0012`. Some something should be fixed.

I was also wondering why the hexadecimal passed whereas this one did not pass, which made me realize that the hex test does not have a test for negative numbers with `%05x`. So we should probably add one like that as well.